[ITBL-1163] in app notifications mobile ui

davidtruong opened the pull request:
In progress work for in app notification. Will finalize once we get the backend hooked up.

davidtruong committed:
Base code for inApp Notifications

davidtruong committed:
Updates deprecated async load method

davidtruong committed:
Minor layout changes
Adds in support for Selector callbacks for the button clicks.

davidtruong committed:
Adds in block callback handlers for button clicks.

davidtruong committed:
Updates the typedef scope for the actionBlock.

davidtruong committed:
Refactored the view controllers into the base class.

Added in support for int hex color types.

davidtruong committed:
Updated ItereableAlertViewController to parse json payload.

davidtruong committed:
code cleanup

davidtruong committed:
Support for multibuttons in the json payload.

davidtruong committed:
Misc cleanup

dontgitit commented on Iterable-iOS-SDK/IterableAlertView.h:
```diff
@@ -0,0 +1,69 @@
+
+#import <UIKit/UIKit.h>
+
+typedef NS_ENUM(NSInteger, IterableAlertViewButtonType) {
```
NSInteger vs NSUInteger in the other enum?

dontgitit replied:
i think you can add docstrings for these somehow

dontgitit commented on Iterable-iOS-SDK/IterableFullScreenViewController.m:
```diff
@@ -0,0 +1,188 @@
+
+@implementation IterableFullScreenViewController
+
+CGFloat imageWidth =0;
```
weird spacing

dontgitit commented on Iterable-iOS-SDK/IterableInAppManager.m:
```diff
@@ -0,0 +1,92 @@
+        baseNotification = [[IterableAlertViewController alloc] initWithNibName:nil bundle:nil];
+    }
+    
+    //TODO: add in view tracking
```
TODO?

jontsai commented on Iterable-iOS-SDK/IterableConstants.h:
```diff
@@ -0,0 +1,34 @@
+//  IterableConstants.h
+//  Iterable-iOS-SDK
+//
+//  Created by David Truong on 9/9/16.
```
my personal pref is to remove comments like this

dontgitit commented:
adds some docs for the public apis, otherwise lftm

davidtruong committed:
Removed access for initWithApiKey in the header. Added in the userId field

davidtruong committed:
Updated api calls to take an optional UserId

davidtruong committed:
fixes disableDevice to allow userIds

davidtruong committed:
Large refactor to support the payload from the inapp Get endpoint.
Added SpawnInAppNotifications: getInAppMessages + showNotification
Support for Get Requests

Change to squared buttons
Removed padding between buttons
Added null checks for payload params
Updated constants to match backend
Changed RGBA support to used fixed alpha level
Changed default values for dialog color/text

davidtruong committed:
Updates full screen 2 button layout
Updates text auto sizing on full screen.

davidtruong committed:
Added error handling for spawnInAppNotification request
Changed logic for getting the next in-app notification
Added documentation for IterableApi
Updated constants values

davidtruong committed:
refactoring buttons inApp

davidtruong committed:
updated spawnInApp api to include a callback function handler
Fixed up cocoaPod lint warnings

davidtruong committed:
cleanup + error checking for alertView InApp

davidtruong committed:
Added in trackINAppView and trackInAppClick stub calls

Documentation of classes/functions

davidtruong committed:
Cleanup and code restructure for InApp

davidtruong committed:
ITBL-1535 Added prefix to common method names
Code review fixes + code cleanup/refactor + documentation

davidtruong committed:
Removing outdated load image function from InApp.

davidtruong committed:
Integrated changes to support InApp Tracking

davidtruong committed:
Adds in messageId to InApp metadata

davidtruong committed:
Adds in count parameter for get InAppMessages.

davidtruong committed:
Added in iOS themed soft-prompt dialogs.

hjz commented on Iterable-iOS-SDK/IterableAPI.h:
```diff
@@ -52,6 +54,12 @@ typedef NS_ENUM(NSInteger, PushServicePlatform) {
 @property(nonatomic, readonly, copy) NSString *email;
```
can we add a comment, that the email is the preferred key

davidtruong replied:
Added the comment in the function and in the README docs.

hjz commented on Iterable-iOS-SDK/IterableAPI.h:
```diff
@@ -63,27 +71,20 @@ typedef NS_ENUM(NSInteger, PushServicePlatform) {
  @return an instance of IterableAPI
  */
-- (instancetype) initWithApiKey:(NSString *)apiKey andEmail:(NSString *)email launchOptions:(nullable NSDictionary *)launchOptions;
++ (IterableAPI *) sharedInstanceWithApiKey:(NSString *)apiKey andUserId:(NSString *)userId launchOptions:(nullable NSDictionary *)launchOptions;
```
Is the email option removed?

davidtruong replied:
It has its own function with a separate call signature. It is still there as: + (IterableAPI *) sharedInstanceWithApiKey:(NSString *)apiKey andEmail:(NSString *)email

andrewboni commented on Iterable-iOS-SDK/IterableAPI.m:
```diff
@@ -86,6 +87,32 @@ - (NSURL *)getUrlForAction:(NSString *)action
+    for (NSString* paramKey in args) {
+        NSString* paramValue = args[paramKey];
+        
+        NSString *params = [NSString stringWithFormat:@"&&%@=%@", paramKey, paramValue];
```
single `&`

davidtruong replied:
done

davidtruong committed:
Merge branch 'master' into feature/ITBL-1163-in-app-notifications-mobile-ui

# Conflicts:
#	Iterable-iOS-SDK/IterableAPI.m

davidtruong committed:
Fixing up merge conflicts with disableDevice

davidtruong committed:
Revamped constants for IterableApi.

davidtruong committed:
Updated api docs
Fix for IterableConstants interface
Fixed up unit tests

davidtruong reviewed (commented):
Made changes

davidtruong committed:
code review changes to add comments and update get request params
